src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -104,6 +97,25 @@ class LedgerImpl(vm: VM, blockchainConfig: BlockchainConfig) extends Ledger with
     blockTxsExecResult
   }
 
+  @tailrec
+  private[ledger] final def executePreparedTransactions(
+    signedTransactions: Seq[SignedTransaction], world: InMemoryWorldStateProxy,
+    blockHeader: BlockHeader, signedTransactionValidator: SignedTransactionValidator,
+    acumGas: BigInt = 0, acumReceipts: Seq[Receipt] = Nil, executed: Seq[SignedTransaction] = Nil)
+  : Either[TxsExecutionError, (BlockResult, Seq[SignedTransaction])] = {
+
+    val result = executeTransactions(signedTransactions, world, blockHeader, signedTransactionValidator, acumGas, acumReceipts)
```

Comment:
I did not want to repeat steps from executeTransactions()
also if I do it in recursive fashion I would need `StateBeforeFailure`